autometrics/src/prometheus_exporter.rs

```diff
@@ -124,12 +153,31 @@ pub fn global_metrics_exporter() -> GlobalPrometheus {
 /// # use http::StatusCode;
 /// // Mounted at the route `/metrics`
 /// pub async fn metrics_get() -> (StatusCode, String) {
-///   match autometrics::encode_global_metrics() {
+///   match autometrics::prometheus_exporter::encode_to_string() {
 ///     Ok(metrics) => (StatusCode::OK, metrics),
 ///     Err(err) => (StatusCode::INTERNAL_SERVER_ERROR, format!("{:?}", err))
 ///   }
 /// }
 /// ```
-pub fn encode_global_metrics() -> Result<String, EncodingError> {
+pub fn encode_to_string() -> Result<String, EncodingError> {
     GLOBAL_EXPORTER.encode_metrics()
 }
+
+/// Export the collected metrics to the Prometheus or OpenMetrics format and wrap
+/// them in an HTTP response.
+///
+/// If you are using exemplars, this will automatically use the OpenMetrics
+/// content type so that Prometheus can scrape the metrics and exemplars.
+pub fn encode_http_response() -> PrometheusResponse {
+    match encode_to_string() {
+        Ok(metrics) => http::Response::builder()
+            .status(200)
+            .header(CONTENT_TYPE, RESPONSE_CONTENT_TYPE)
+            .body(metrics)
+            .expect("Error building response"),
```

Comment:
we should consider using `Result`s instead of panic-ing, especially because this will be used by long running http servers

Reply:
That's true, but I think this can only fail if one of the things we pass in is invalid https://docs.rs/http/latest/http/response/struct.Builder.html#errors